docs/eeps/draft-maven-style-plugins.md

```diff
@@ -0,0 +1,246 @@
+
+The other way to configure with Maven-style plugins is Embulk System Properties. See [EEP-4](./eep-0004.md) for Embulk System Properties.
+
+Once a user sets an Embulk System Property `plugins.<category>.<type>` in the format of `maven:<group>:<name>:<version>`, the same YAML configuration as RubyGems-style keeps working. The following examples configure the same with the above to run with the S3 Input Plugin of the Maven-style plugin artifact `org.embulk:embulk-input-s3:0.5.3`.
+
+```
+# embulk.properties
+plugins.input.s3=maven:org.embulk:s3:0.5.3
+```
+
+```
+# maven-with-properties.yaml
+in:
+  type: s3
+  bucket: ...
+  path_prefix: ...
+```
+
+The `artifactId` is not fully specified like `embulk-input-s3` there like the inline configuration explained above.
+
+The latter, with Embulk System Properties, is expected to be less confusing for users.
+
+### Type conflicts
+
+The older RubyGems-style plugins can be installed along with the newer Maven-style plugins. It means they conflict. When a user specifies `type` of a plugin that has a conflict between the RubyGems-style and the Maven-style, the priority is determined as follows.
+
+* The Maven-style plugin is prioritized if the corresponding Embulk System Property `plugins.<category>.<type>` is configured, and the Maven artifact with the specified `groupId` and `version` is installed.
+* Otherwise, the RubyGems-style plugin is chosen by definition.
+
+Main class and Manifest
+------------------------
+
+Once the plugin's installed JAR artifact is determined from the user's YAML configuration, its dependencies are identified from `pom.xml` of the plugin artifact. Only the direct dependencies of the plugin artifact are included, as explained in "Intended limitation for stable dependency resolution".
+
+These JAR files are the "classpath" to be loaded for the Maven-style plugin. Embulk can load these classes now, but Embulk still needs to identify the "main class" as an Embulk plugin, which implements Embulk SPI's `*Plugin` interface, such as `org.embulk.spi.InputPlugin`, `org.embulk.spi.FilterPlugin`, and else.
+
+The main class is to be declared in `META-INF/MANIFEST.MF`, [the manifest file](https://docs.oracle.com/javase/tutorial/deployment/jar/manifestindex.html) of the plugin JAR file. The manifest file of an Embulk plugin's JAR file must include the following attributes, `Embulk-Plugin-Spi-Version`, `Embulk-Plugin-Category`, `Embulk-Plugin-Main-Class`, and `Embulk-Plugin-Type`.
+
+* `Embulk-Plugin-Spi-Version` must be `0` as of Embulk v0.11 or earlier.
+* `Embulk-Plugin-Category` is the category of the plugin, which is one of `input`, `decoder`, `parser`, `filter`, `formatter`, `encoder`, `output`, or `guess`.
+* `Embulk-Plugin-Main-Class` is the "main class" of the Embulk plugin, which implements Embulk SPI's `*Plugin` interface.
+* `Embulk-Plugin-Type` is the type of the Embulk plugin, such as `file`, `s3`, `csv`, `gzip`, or else.
+
+See the following example of a manifest file.
```

Comment:
How about writing a descritpon about Non `Embulk-` prefix attributes? (I a bit confused about `Implementation-Title: embulk-input-file` )
ex) Non `Embulk` prefix attribute description are described in the manifest file.

Reply:
They're Java's standard ones, and explained in a Java page linked from the URL above.

https://docs.oracle.com/javase/tutorial/deployment/jar/packageman.html

Reply:
Added a quick note:

> Note that `Manifest-Version`, `Implementation-Title`, and `Implementation-Version` are Java standard ones. See ["Working with Manifest Files: The Basics"](https://docs.oracle.com/javase/tutorial/deployment/jar/manifestindex.html) for more details.